proxy_agent/src/key_keeper.rs

```diff
@@ -519,22 +519,23 @@ mod tests {
             20,
         );
 
+        let shared_state = SharedState::new();
         // start wire_server listener
         let ip = "127.0.0.1";
         let port = 8081u16;
-
-        // LATER: need switch to tokio spawn once the server_mock is async
-        std::thread::spawn(move || {
-            server_mock::start(ip.to_string(), port);
-        });
+        let cloned_shared_state = shared_state.clone();
```

Comment:
Is this redundant? It gets cloned on final usage as well

Reply:
good catch.